Game/MyProject/Game/Helper/HelperManager.cpp

```diff
@@ -0,0 +1,553 @@
+        node->setRenderable(meshInstance);
+        
+        return node;
+    }
+    //---------------------------------------------------------------------------
+    // @HelperManager::BuildQuad()
+    //---------------------------------------------------------------------------
+    std::shared_ptr<SceneNode>
+    BuildQuad(const std::shared_ptr<MeshInstance>&meshInstance,IvVector3 axis,bool wireframeValue)
+    {
+        RenderPacket myRenderPacket;
+        myRenderPacket._use_blend = true;
+        myRenderPacket._use_depth = true;
+        myRenderPacket._use_wireframe = wireframeValue;
+        myRenderPacket._prim_type = kTriangleStripPrim;
+        
+        std::shared_ptr<HelperSceneNode> node = std::make_shared<HelperSceneNode>("node", myRenderPacket);
+        node->setRenderable(meshInstance);
+        
+        IvVector3 position=IvVector3(0,0,0);
+        IvVector3 rotation=IvVector3(0,0,0);
+        IvVector3 scale=IvVector3(1,1,1);
+        if (axis == IvVector3::xAxis)
+            rotation += IvVector3(0, 1.5, 0);
+        else if (axis == IvVector3::yAxis)
+            rotation += IvVector3(1.5, 0, 0);
+        else if (axis == IvVector3::zAxis)
+            rotation += IvVector3(0, 0, 1.5);
+        node->setLocalTransform(position, rotation, scale);
+        return node;
+    }
+    
+    //---------------------------------------------------------------------------
+    // @HelperManager::BuildTexturedQuad()
+    //---------------------------------------------------------------------------
+    std::shared_ptr<SceneNode>
+    BuildTexturedQuad(const std::shared_ptr<MeshInstance>&meshInstance,const char* textureName,const char* shaderName,IvVector3 axis,bool wireframeValue)
+    {
+        RenderPacket myRenderPacket;
+        myRenderPacket._use_blend = true;
+        myRenderPacket._use_depth = true;
+        myRenderPacket._use_wireframe = wireframeValue;
+        myRenderPacket._prim_type = kTriangleStripPrim;
+        
+        std::shared_ptr<HelperSceneNode> node = std::make_shared<HelperSceneNode>("node", myRenderPacket);
+        node->setRenderable(meshInstance);
+        
+        IvVector3 position=IvVector3(0,0,0);
+        IvVector3 rotation=IvVector3(0,0,0);
+        IvVector3 scale=IvVector3(1,1,1);
+        if (axis == IvVector3::xAxis)
+            rotation += IvVector3(0, 1.5, 0);
+        else if (axis == IvVector3::yAxis)
+            rotation += IvVector3(1.5, 0, 0);
+        else if (axis == IvVector3::zAxis)
+            rotation += IvVector3(0, 0, 1.5);
+        node->setLocalTransform(position, rotation, scale);
+        
+        IvImage* image = IvImage::CreateFromFile(textureName);
+        IvTexture* quadTexture;
+        
+        if (image)
+        {
+            quadTexture = IvRenderer::mRenderer->GetResourceManager()->CreateTexture((image->GetBytesPerPixel() == 4) ? kRGBA32TexFmt : kRGB24TexFmt, image->GetWidth(), image->GetHeight(), image->GetPixels(), kImmutableUsage);
```

Comment:
too much stuff going on in this line of code, break stuff up so it's readable. see the code smell url above